physics/GFS_surface_composites.F90

```diff
@@ -256,39 +258,38 @@ subroutine GFS_surface_composites_pre_run (im, flag_init, flag_restart, lkm, fra
         endif
       enddo
 !
-      if (.not. cplflx .or. kdt == 1) then
-        if (frac_grid) then
-          do i=1,im
-            if (dry(i)) then
-              if (icy(i)) then
+      if (frac_grid) then
+        do i=1,im
+          if (dry(i)) then
+            if (icy(i)) then
+              if (kdt == 1 .or. (.not. cplflx .or. lakefrac(i) > zero)) then 
                 tem = one / (cice(i)*(one-frland(i)))
                 snowd_ice(i) = max(zero, (snowd(i) - snowd_lnd(i)*frland(i)) * tem)
                 weasd_ice(i) = max(zero, (weasd(i) - weasd_lnd(i)*frland(i)) * tem)
               endif
-            elseif (icy(i)) then
+            endif
+          elseif (icy(i)) then
+            if (kdt == 1 .or. (.not. cplflx .or. lakefrac(i) > zero)) then 
               tem = one / cice(i)
               snowd_lnd(i) = zero
               snowd_ice(i) = snowd(i) * tem
               weasd_lnd(i) = zero
               weasd_ice(i) = weasd(i) * tem
             endif
-          enddo
-        else
-          do i=1,im
-            if (dry(i)) then
-              snowd_lnd(i) = snowd(i)
-              weasd_lnd(i) = weasd(i)
-              snowd_ice(i) = zero
-              weasd_ice(i) = zero
-            elseif (icy(i)) then
+          endif
+        enddo
+      else
+        do i=1,im
+          if (icy(i)) then
+            if (kdt == 1 .or. (.not. cplflx .or. lakefrac(i) > zero)) then
```

Comment:
@SamuelTrahanNOAA A PR#131 has to be merged into this code. This line should be:
if ((kdt == 1.and. lsm_cold_start) .or. (.not. cplflx .or. lsm==lsm_ruc .or. lakefrac(i) > zero)) then